src/seed_tools/utils/studies_to_seed.ts

```diff
@@ -48,20 +51,68 @@ async function readStudiesFromDirectory(
 }> {
   const files = (await fs.readdir(studiesDir)).sort();
 
+  const filesWithContent = [];
+  for (const file of files) {
+    const filePath = path.join(studiesDir, file);
+    const content = await fs.readFile(filePath, 'utf8');
+    filesWithContent.push({ path: filePath, content });
+  }
+
+  return await readStudies(filesWithContent, fix);
+}
+
+async function readStudiesAtRevision(
+  studiesDir: string,
+  revision: string,
+): Promise<{
+  studies: Study[];
+  studyFileBaseNameMap: Map<Study, string>;
+  errors: string[];
+}> {
+  const basePath = wsPath('//');
+  studiesDir = path.relative(basePath, studiesDir);
+  const files = execSync(`git show ${revision}:${studiesDir}`, {
+    encoding: 'utf8',
+  }).split('\n');
+
+  const filesWithContent = [];
+  for (const file of files) {
+    if (!file.endsWith('.json5')) continue;
+    const content = execSync(`git show ${revision}:"${studiesDir}/${file}"`, {
```

Comment:
looked through this one in a bit more detail and it seems that it's been refactored since https://github.com/brave/brave-variations/pull/1202/ which I originally checked and didn't see issues with, but now I'm seeing how this could be user controlled (e.g. whoever executes the command can control this)

Looking at this it does seem like this should be filtered to prevent issues.

Reply:
@kdenhartog 
yep, it's refactoring, not a new issue.
The command is called from a command line: `npm run seed_tools create -- <studyDirectory> --revision <sha1>`

Not sure we have sanitize args from a command line.

Reply:
Ok, that's what I was thinking. In this case if you can modify the args passed into the command line you might as well just execute your own commands. Not too concerned with these.

If I was to suggest a way to sanitize these it would be to just check the values [here](https://github.com/brave/brave-variations/pull/1245/files#diff-e4dd7f569a9d2e49e70d5212a0f2ca4799d321d8d47db86b94c59a404e92e714R52-R54) before we pass them in further to make sure they're expected. E.g. revision parameter should be a hash and studyDir we could probably check to make sure it's within a reasonable location on the file system and matches a file path (rather than appending on something like `&& npm run malicious script` or something to that affect.